src/ddspdrum/module.py

```diff
@@ -268,10 +245,47 @@ def __call__(self, envelopecontrol: np.array, audiosample: np.array):
         envelopecontrol = np.clip(envelopecontrol, 0, 1)
         audiosample = np.clip(audiosample, -1, 1)
         amp = self.control_to_sample_rate(envelopecontrol)
-        signal = self.fix_length(audiosample, len(amp))
+        signal = fix_length(audiosample, len(amp))
         return amp * signal
 
 
+class Drum:
+    """
+    A package of modules that makes one drum hit. 
+    TODO:   - clicks when the envelopes don't match up.
+    """
+
+    def __init__(
+        self,
```

Comment:
I think it would be cleaner and safer to construct this by passing in:

```
pitch_adsr: ADSR, amp_adsr: ADSR, vco: VCO, vca: VCA
```

Otherwise it's easy to fuck up the parameters

Reply:
I agree this is weird, but the idea is the drum module does all the work for you, otherwise you might as well build it from scratch, no?

Reply:
nm, it's all good I did it.